Plugins/translate/TranslatePlugin.cs

```diff
@@ -0,0 +1,73 @@
+﻿using BasePlugin;
+using BasePlugin.Interfaces;
+using BasePlugin.Records;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Text.Json;
+
+namespace List
+{
+    public class TranslatePlugin : IPlugin
+    {
+        public const string _Id = "translate";
+        public string Id => _Id;
+        private static readonly string subscriptionKey = Environment.GetEnvironmentVariable("TranslateKey");
+        private static readonly string endpoint = "https://api.cognitive.microsofttranslator.com/";
+        private static readonly string location = "eastus2";
+
+        public async void Execute(PluginInput input)
+        {
+            if (input.Message == "")
+            {
+                input.Callbacks.StartSession();
+                input.Callbacks.SendMessage("Type a sentence to translate. Enter 'len' to which language to translate into. Enter 'Exit' to stop");
+            }
+            else if (input.Message.ToLower() == "exit")
+            {
+                input.Callbacks.EndSession();
+                input.Callbacks.SendMessage("Out of translation.");
+            }
+            else if (input.Message.ToLower().StartsWith("len"))
+            {
+                var str = input.Message.Substring("len".Length).Trim();
+                input.Callbacks.SavePluginUserData(str);
+
+                input.Callbacks.SendMessage($"the translation language: {str}");
+            }
+            else
+            {
+                translate(input.Message, input).Wait();
+            }
+        }
+        static async Task translate(string textToTranslate, PluginInput input)
+        {
+            // Input and output languages are defined as parameters.
+            string len = input.PersistentData;
+            string route = "/translate?api-version=3.0&from=&to="+ len + "&to=it";
+            object[] body = new object[] { new { Text = textToTranslate } };
+            var requestBody = JsonConvert.SerializeObject(body);
+
+            using (var client = new HttpClient())
+            using (var request = new HttpRequestMessage())
+            {
+                // Build the request.
+                request.Method = HttpMethod.Post;
+                request.RequestUri = new Uri(endpoint + route);
+                request.Content = new StringContent(requestBody, Encoding.UTF8, "application/json");
+                request.Headers.Add("Ocp-Apim-Subscription-Key", subscriptionKey);
+                request.Headers.Add("Ocp-Apim-Subscription-Region", location);
+
+                // Send the request and get response.
+                HttpResponseMessage response = await client.SendAsync(request).ConfigureAwait(false);
+                // Read response as a string.
+                string result = await response.Content.ReadAsStringAsync();
+                input.Callbacks.SendMessage(result.Split(':')[5].Split(',')[0].Split('"')[1]);
```

Comment:
For code separation
This function should return a string instead of sending the message directly